### PR TITLE
1546 changing meta data malfunctions

### LINKED ISSUE #1546: Importer: Changing a mapped metaData results in error
Steps to reproduce:

1. Open a file (eg in the case of this screenshot [Data.xlsx](https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/files/8242860/Data.xlsx) )
2. Select for "Route" one mapping, fe excel column Group.
3. Change the selection  for "Route" to another mapping, fe excel column Route.
The new selection results in the following error:

image

### PR DESCRIPTION
**Pull request by georgeDaskalakis**
closes #1546 

@msevestre sorry, running the ReSharper formatter makes the review a bit harder. But at least the formating in the files is correct that way :)

### CODE REVIEW

File: src/OSPSuite.Presentation/Presenters/Importer/ColumnMappingPresenter.cs [hunk 1]
```diff
@@ -523,14 +533,15 @@ public IEnumerable<ColumnMappingOption> GetAvailableOptionsFor(ColumnMappingDTO
...
-      private IEnumerable<string> 
+      private IEnumerable<string>
          availableColumns()
       {
          return _format.ExcelColumnNames
             .Where
             (
                cn =>
-                  _format.Parameters.OfType<MappingDataFormatParameter>().All(p => p.ColumnName != cn && p.MappedColumn?.Unit?.ColumnName != cn && p.MappedColumn?.LloqColumn != cn) &&
+                  _format.Parameters.OfType<MappingDataFormatParameter>().All(p =>
```
Comment (msevestre) [inline src/OSPSuite.Presentation/Presenters/Importer/ColumnMappingPresenter.cs:543]: This statement needs to be rewritten at some point. I have absolutely NO idea what this means. So many ANDs, NOT, without comment. Let's be kind to own future selves and not do that anymore

**Comment by msevestre** [comment]
Unfortuntely I do not see what has changed that is related to the actual issue at hand. When so much formatting is going on, it would be helpful for me if you could point out the actual changes. So I am going to ask for revision for now until I see what I need to review.

**Review by msevestre — COMMENTED** [review]
I cannot review this changes at the moment as running resharper changed way too much. That's why running resharper before each PUSH is something we should try to do

File: src/OSPSuite.Presentation/Presenters/Importer/ColumnMappingPresenter.cs (hunk 1 above)
Comment (georgeDaskalakis) [reply src/OSPSuite.Presentation/Presenters/Importer/ColumnMappingPresenter.cs:543]: OK, I ll add a todo here for now and we can talk about refactoring such code.

**Comment by georgeDaskalakis** [comment]
> I cannot review this changes at the moment as running resharper changed way too much. That's why running resharper before each PUSH is something we should try to do

@msevestre  you mean doing a PR just with the format changes seperately right? Yeap, I agree with that.

File: tests/OSPSuite.Presentation.Tests/Importer/Presenters/ColumnMappingPresenterSpecs.cs [hunk 2]
```diff
@@ -545,6 +562,108 @@ public void the_unit_and_dimension_are_set_properly(string oldUnitDescription, s
...
+      }
+
+      [Observation]
+      public void the_meta_data_row_should_be_correctly_added()
+      {
+         var metaDataFormat = (_secondModel.Source as MetaDataFormatParameter);
+         metaDataFormat.ColumnName.ShouldBeEqualTo("Patient id");
+      }
+   }
+
```
Comment (georgeDaskalakis) [inline tests/OSPSuite.Presentation.Tests/Importer/Presenters/ColumnMappingPresenterSpecs.cs:666]: @msevestre in this file, only the tests written in these lines here are new

File: src/OSPSuite.Presentation/Presenters/Importer/ColumnMappingPresenter.cs [hunk 3]
```diff
@@ -572,45 +584,57 @@ private void setUnitAndDimension(ColumnMappingDTO model)
...
+                     break;
+                  default:
+                     throw new NotImplementedException($"Setting description for unhandled column type: {model.CurrentColumnType}");
+               }
+
+               break;
             }
-            setUnitAndDimension(model);
          }
       }
```
Comment (georgeDaskalakis) [inline src/OSPSuite.Presentation/Presenters/Importer/ColumnMappingPresenter.cs:640]: @msevestre here it is only this function SetDescriptionForRow that has been changed. The actual fix here is the moving setUnitAndDimension(model) call to the correct place. And I have also refactored a bit the switch/if...then..else statements  according to ReSharper suggestions.

**Comment by georgeDaskalakis** [comment]
> Unfortuntely I do not see what has changed that is related to the actual issue at hand. When so much formatting is going on, it would be helpful for me if you could point out the actual changes. So I am going to ask for revision for now until I see what I need to review.

@msevestre I have made comments on the 2 places that the code has really changed. Not sure if it is easier for you that I marked the whole code, or if it generally easier to mark just the first 2-3 lines and explain, but please give me feedback so that I can do in a way that works better for you next time :)

**Comment by georgeDaskalakis** [comment]
@msevestre I did not refactor the function that we discussed yesterday, I am in the middle of that bigger refactoring with DataSheetCollection, and I think I cannot approach both concurrently with the necessary caution. I have though left a todo there for the refactoring, along with detailed comments of what every case in the switch does. But this PR should be ready now. 

We could open an issue where we kind of put together things that we have to refactor, what do you think?

**Comment by msevestre** [comment]
>We could open an issue where we kind of put together things that we have to refactor, what do you think?

yes please. Can you do it? Merging this

**Pull request merged**